src/output-tx.c

```diff
@@ -71,20 +74,60 @@ int OutputRegisterTxLogger(const char *name, AppProto alproto, TxLogger LogFunc,
     op->output_ctx = output_ctx;
     op->name = name;
     op->module_id = (TmmId) module_id;
+    op->tc_log_progress = 0;
+    op->ts_log_progress = 0;
 
-    if (list == NULL)
+    if (list == NULL) {
+        op->id = 1;
         list = op;
-    else {
+    } else {
         OutputTxLogger *t = list;
         while (t->next)
             t = t->next;
+        op->id = t->id * 2;
```

Comment:
we need a check to make sure this value is within the limits of the bitarray. I think you've used 32 bits so we should make sure it doesn't exceed that.